storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1469,974 @@ public static int toPositive(int number) {
+        if(m.containsKey(key)) {
+            HashMap<Object, String> newMap = new HashMap<>(m);
+            String value = newMap.get(key);
+            String redacted = new String(new char[value.length()]).replace("\0", "#");
+            newMap.put(key, redacted);
+            return newMap;
+        }
+        return m;
+    }
+
+    public static void logThriftAccess(Integer requestId, InetAddress remoteAddress, Principal principal, String operation) {
+        new ThriftAccessLogger().log(
+            String.format("Request ID: {} access from: {} principal: {} operation: {}",
+                          requestId, remoteAddress, principal, operation));
+    }
+
+    /**
+     * Make sure a given key name is valid for the storm config. 
+     * Throw RuntimeException if the key isn't valid.
+     * @param name The name of the config key to check.
+     */
+    public static void validateKeyName(String name) {
+        Set<String> disallowedKeys = new HashSet<>();
+        disallowedKeys.add("/");
+        disallowedKeys.add(".");
+        disallowedKeys.add(":");
+        disallowedKeys.add("\\");
+
+        for(String key : disallowedKeys) {
+            if( name.contains(key) ) {
+                throw new RuntimeException("Key name cannot contain any of the following: " + disallowedKeys.toString());
+            }
+        }
+        if(name.trim().isEmpty()) {
+            throw new RuntimeException("Key name cannot be blank");
+        }
+    }
+
+    //Everything from here on is translated from the old util.clj (storm-core/src/clj/backtype.storm/util.clj)
+
+    public static final boolean IS_ON_WINDOWS = "Windows_NT".equals(System.getenv("OS"));
+
+    public static final String FILE_PATH_SEPARATOR = System.getProperty("file.separator");
+
+    public static final String CLASS_PATH_SEPARATOR = System.getProperty("path.separator");
+
+    public static final int SIGKILL = 9;
+    public static final int SIGTERM = 15;
+
+
+
+    /**
+     * Find the first item of coll for which pred.test(...) returns true.
+     * @param pred The IPredicate to test for
+     * @param coll The Collection of items to search through.
+     * @return The first matching value in coll, or null if nothing matches.
+     */
+    public static Object findFirst (IPredicate pred, Collection coll) {
+        if (coll == null || pred == null) {
+            return null;
+        } else {
+            Iterator<Object> iter = coll.iterator();
+            while(iter != null && iter.hasNext()) {
+                Object obj = iter.next();
+                if (pred.test(obj)) {
+                    return obj;
+                }
+            }
+            return null;
+        }
+    }
+
+    public static Object findFirst (IPredicate pred, Map map) {
+        if (map == null || pred == null) {
+            return null;
+        } else {
+            Iterator<Object> iter = map.entrySet().iterator();
```

Comment:
Here too using Iterator directly just makes things more complicated.

```
public static <K, V> Map.Entry<K, V> findFirst (IPredicate pred, Map<K, V> map) {
  if (map == null) {
    return null;
  }
  return findFirst(pred, map.entrySet());
}
```

Oh and it would be nice to have IPredicate be a generic as well for the Parameter passed in.